DndEncounterSimulator/Objects/Creature.py

```diff
@@ -52,6 +53,17 @@ def damage(self, damages: int):
         if self.hit_points <= 0:
             self.dead = True
 
+    def change_weapon(self, index: int):
+        """
+        Method to change to the weapon references by the index : this weapon will be placed at index 0 for use in combat
+        :param index: (int) the index referencing a weapon in weapons list

```

Comment:
Docstring conventions: leave a blank line between description & details (here, your parameters).

+ I'd say "referenced", and there is no space between ":" in English
```suggestion
        Method to change to the weapon referenced by the index: this weapon will be placed at index 0 for use in combat
        
        :param index: (int) the index referencing a weapon in weapons list

```